### ISSUE TITLE
Loading line item extension from custom cdn behind firewall

### ISSUE THREAD

**Issue by plamenbotev** [SharePoint/sp-dev-docs#9262]
### What type of issue is this?

Question

### What SharePoint development model, framework, SDK or API is this about?

💥 SharePoint Framework

### Target SharePoint environment

SharePoint Online

### What browser(s) / client(s) have you tested

- [ ] 💥 Internet Explorer
- [ ] 💥 Microsoft Edge
- [ ] 💥 Google Chrome
- [ ] 💥 FireFox
- [ ] 💥 Safari
- [ ] mobile (iOS/iPadOS)
- [ ] mobile (Android)
- [ ] not applicable
- [ ] other (enter in the "Additional environment details" area below)

### Additional environment details

- browser version
- SPFx version
- Node.js version
- etc

### Issue description

Hello,

Is it possible to fail early, when trying to load the js of a line item extension from a custom cdn, that is fire walled?
The idea is to not block the list of line items if the asset can not be loaded for some users and only allow the extension to load for certain users, that are white listed? 
It looks like this was working before, but now the whole list of line items is blocked until the assets are loaded and thus for some users the list is unusable. 

Best Regards

**Comment by nick-pape** [comment]
Absolutely, we will take a look at this. The list of items should still be rendered even if an extension fails to load.

Are you doing this on a regular SharePoint list, or is the issue happening elsewhere (e.g. a list Web Part)?

**Comment by ivaylomitrev** [comment]
Hey,

Has this been confirmed as a bug and, if so, would you happen to know what the timelines for fixing it would be?

**Comment by nick-pape** [comment]
Hi there @ivaylomitrev ,

I talked this over with the team, and this is by-design behavior. For better or worse, it's a better practice for us to fail to render the list item to give the developer/user some feedback that the list item was not rendered, rather than to fail silently if the list item extension failed to load.

However, there should be some UX for this. Can you share a screenshot of what you are seeing in the list UX when this issue occurs?

**Comment by plamenbotev** [comment]
Hello @nick-pape and thank you for the feedback.
I will attach additionally screenshots.
At the moment, the whole list / library is not accessible if there is  a line item extension that can not be loaded.
I think that if there is a timeout during loading the extension assets, the client and the developer can be notified in a way that will keep the list still usable and the data visible. 
The user could be additionally notified by flash message for example.

**Comment by plamenbotev** [comment]
Hello again @nick-pape,
Just checking if there is any new information regarding this issue? :) 

Best Regards,
Plamen

**Comment by Ashlesha-MSFT** [comment]
@plamenbotev,
Just checking in to see if the issue has been resolved?

If you're still experiencing the problem, could you please provide detailed repro steps so we can further investigate the behavior? Specifically, it would help to know:

- The extension type you're using (e.g., Command Set or Field Customizer)
- Where it's deployed (CDN details if possible)
- How the access restriction is configured (firewall/whitelisting)
- Steps to reproduce the issue in your environment

Looking forward to your update!

**Comment by Ashlesha-MSFT** [comment]
We are closing this issue for now due to inactivity from the developer. If the problem persists or reappears, feel free to reopen this issue or open a new one.
